core/crypto.mk

```diff
@@ -192,20 +192,35 @@ endif
 ifeq ($(CFG_CRYPTOLIB_NAME),mbedtls)
 # mbedtls has to be complemented with some algorithms by LTC
 # Specify the algorithms here
-_CFG_CORE_LTC_DSA := $(CFG_CRYPTO_DSA)
-_CFG_CORE_LTC_MPI := $(CFG_CRYPTO_DSA)
-_CFG_CORE_LTC_SHA256_DESC := $(CFG_CRYPTO_DSA)
-_CFG_CORE_LTC_SHA384_DESC := $(CFG_CRYPTO_DSA)
-_CFG_CORE_LTC_SHA512_DESC := $(CFG_CRYPTO_DSA)
+ifeq ($(CFG_CRYPTO_DSA),y)
+_CFG_CORE_LTC_DSA := y
+_CFG_CORE_LTC_MPI := y
+_CFG_CORE_LTC_SHA256_DESC := y
+_CFG_CORE_LTC_SHA384_DESC := y
+_CFG_CORE_LTC_SHA512_DESC := y
+endif
 _CFG_CORE_LTC_XTS := $(CFG_CRYPTO_XTS)
 _CFG_CORE_LTC_CCM := $(CFG_CRYPTO_CCM)
-_CFG_CORE_LTC_AES_DESC := $(call cfg-one-enabled, CFG_CRYPTO_XTS CFG_CRYPTO_CCM)
+ifneq (,$(filter y, $(CFG_CRYPTO_XTS) $(CFG_CRYPTO_CCM)))
+_CFG_CORE_LTC_AES_DESC := y
+endif
 endif
 
 ###############################################################
 # libtomcrypt (LTC) specifics, phase #2
 ###############################################################
 
+_CFG_CORE_LTC_SHA256_DESC := $(call cfg-one-enabled, _CFG_CORE_LTC_SHA256_DESC \
+						     _CFG_CORE_LTC_SHA224 \
+						     _CFG_CORE_LTC_SHA256)
```

Comment:
@jforissier what's your take on this?

Reply:
I suppose the question is about the use of `:=` (vs. the usual `?=` or `$(call force,,)` right?

IMO it is perfectly fine because it is an "internal" config variable (`_CFG`), so no one is supposed to set it from outside.